### ISSUE TITLE
[Keyboard Navigation- Make code - new branch name]: Screen reader doesn't announce the displayed error message for edit field under "New branch name" dialog.

### ISSUE THREAD

**Issue by Yamini234u** [microsoft/pxt-microbit#5498]
### **User Experience**
Screen reader dependent users will be impacted. If Screen reader is not announcing the error message information which appears for edit field. Due to this issue end users will be confused and unable to get the error message.

**Note:**
User credentials should NOT be included in the bug.
### **Repro Steps:**

1.  Launch application URL : https://makecode.microbit.org
2. Home page will open.
3. Navigate "New Project" option and select it.
4. Editor screen will open.
5. Navigate to "Review, Commit and push local changes to GitHub" button and select it.
6. Navigate through all the controls appearing and select Project name#master Drop down. and select.
7. Navigate through all the controls appearing under "Switch to a different branch" and select "Create new branch".
8. Navigate through all the controls under "New Branch name" Dialogue.
9. Provide the special character in edit field and error message displays
10. Observe and verify whether screen reader announces displayed error message or not.

### **Actual Result:**
 Screen reader doesn't announce the displayed error information for edit field.
**Observation:** 
- On providing invalid input as "%&" error message is displayed but screen reader remains silent.
- Screen reader doesn't announce the infromation "Name cannot have spaces or special characters. Examples: my_feature, add-colors, fix_something" when focus moves to edit field.
**Other Screen reader behavior:** NVDA and Narrator remains silent after error message displayed on the screen.
### **Expected result:** 
The screen reader should announce the displayed error information for edit field.
**Solution:** Screen reader should announce like Error "Don't use spaces or special characters".

### **MAS Reference:**
[MAS 4.1.3 – Status Messages](https://aka.ms/MAS4.1.3)

**ARIA Authoring Ref Link/Fluent Patterns Guidance /HITS Guidance:**
NA

### **DOM details available:**
Yes(Please refer to the attachment)

### **Reference Links:**

- [Accessibility Insights!](https://accessibilityinsights.io/) - Identify accessibility bugs before check-in and make bug fixing faster and easier.
- External Bug Process: If this bug belongs to external team, mark it as resolved/done and assign it back to the tester with notes on where to file/route the bug. For more information, please use this [Link](https://microsoft.sharepoint.com/teams/cea11y/SitePages/External-Bug-Process.aspx?xsdata=MDN8MDF8fDc3NDY0MzExZDkxNTRlYmRhNzk2YTExYzE5Y2I2N2ZifDcyZjk4OGJmODZmMTQxYWY5MWFiMmQ3Y2QwMTFkYjQ3fDF8MHw2Mzc3ODcwNDg4MTQ1NjM2NzR8R29vZHxWR1ZoYlhOVFpXTjFjbWwwZVZObGNuWnBZMlY4ZXlKV0lqb2lNQzR3TGpBd01EQWlMQ0pRSWpvaVYybHVNeklpTENKQlRpSTZJazkwYUdWeUlpd2lWMVFpT2pFeGZRPT0%3D&sdata=VHBCSFBzMTdmWExVNTk3MzZlTHdiME4rcFo3d3lyYk5kTTIvRjhLdTN3QT0%3D&ovuser=72f988bf-86f1-41af-91ab-2d7cd011db47%2cv-chbhardwaj%40microsoft.com). “HCL Staff should not log any third-party external bugs. Should be routed to EDAD team”.
- Please reach out to [C&AI Teams channel](https://nam06.safelinks.protection.outlook.com/?url=https://teams.microsoft.com/l/team/19%253a3cd071821b694f11b054d878453cd75f%2540thread.skype/conversations?groupId%3D30e3f968-15dd-4da2-bd8e-85f27e4af2c3%26tenantId%3D72f988bf-86f1-41af-91ab-2d7cd011db47&data=02%7c01%7cv-dhti%40microsoft.com%7cf90880a914be401871f108d763233116%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c637086876782361405&sdata=/ULS%2BbeqMjysCoiWHvFgAbSnuFJqxO3wtpHmlidPjbU%3D&reserved=0) for any process related queries.

### **Test Environment:**

- **OS Version**: Windows 11 23H2 (OS Build:22631.3007)
- **Browser:** Chrome Version 121.0.2277.4 (Official build) dev (64-bit).
- **Test Matrix:** Chrome + JAWS
- **URL:** https://makecode.microbit.org/.
<img width="964" alt="MAS4 1 3-DOM-Screen reader doesn't announce the displayed error message for edit field under New branch namedialog" src="https://github.com/microsoft/pxt-microbit/assets/117158108/0936d4cd-658c-4d70-8967-2a25ed831db9">

https://github.com/microsoft/pxt-microbit/assets/117158108/ec7a965c-4c4d-47d7-bd55-f9080523e903

**Comment by Yamini234u** [comment]
#Accessibility;#A11yMAS;#A11ySev2;#A11yWCAG2.1;#HCL;#E2E-HCL-MakeCode-Feb2024;#Benchmark;#Win11-Chrome;#MAS4.1.3
